components/StartEndCollection.vue

```diff
@@ -21,53 +21,39 @@
 <script>
 // TODO Validation - end date > start date, no stupidly long events (3 days?), no overlapping dates, only one blank slot.
 // TODO This code could do with a bit of extra testing - various combinations of add/edit/cancel/delete etc.
-const StartEndDate = () => import('~/components/StartEndDate')
+import StartEndDate from '~/components/StartEndDate'
 
 export default {
   components: {
     StartEndDate
   },
   props: {
-    dates: {
+    value: {
       type: Array,
       required: true
     }
   },
-  data: function() {
-    return {
-      editableDates: []
+  async mounted() {
+    if (this.value.length === 0) {
+      this.value.push({
+        uniqueid: await this.$store.dispatch('uniqueid/generate'),
+        start: null,
+        end: null,
+        past: false
+      })
     }
   },
-  mounted: function() {
-    this.editableDates = this.dates ? this.dates : []
-  },
   methods: {
-    remove(uniqueid) {
-      for (let i = 0; i < this.editableDates.length; i++) {
-        if (this.editableDates[i].uniqueid === uniqueid) {
-          this.editableDates.splice(i, 1)
-        }
-      }
-
-      this.$emit('change', JSON.stringify(this.editableDates))
+    remove(idx) {
+      this.value.splice(idx, 1)
```

Comment:
we don't need to use that uniqueid everywhere, in this case, the array index is perfectly sufficient